web/src/main/java/org/cbioportal/web/util/ClinicalDataBinUtil.java

```diff
@@ -0,0 +1,351 @@
+public class ClinicalDataBinUtil {
+
+    @Autowired
+    private StudyViewFilterApplier studyViewFilterApplier;
+    @Autowired
+    private ClinicalDataFetcher clinicalDataFetcher;
+    @Autowired
+    private DataBinner dataBinner;
+    @Autowired
+    private StudyViewFilterUtil studyViewFilterUtil;
+    @Autowired
+    private ClinicalAttributeService clinicalAttributeService;
+    @Autowired
+    private ClinicalAttributeUtil clinicalAttributeUtil;
+    @Autowired
+    private PatientService patientService;
+    
+    public List<ClinicalDataBin> fetchClinicalDataBinCounts(
+        DataBinMethod dataBinMethod,
+        ClinicalDataBinCountFilter dataBinCountFilter
+    ) {
+        List<ClinicalDataBinFilter> attributes = dataBinCountFilter.getAttributes();
+        StudyViewFilter studyViewFilter = dataBinCountFilter.getStudyViewFilter();
+
+        if (attributes.size() == 1) {
+            studyViewFilterUtil.removeSelfFromFilter(attributes.get(0).getAttributeId(), studyViewFilter);
+        }
+
+        List<String> attributeIds = attributes.stream().map(ClinicalDataBinFilter::getAttributeId).collect(Collectors.toList());
+
+        // filter only by study id and sample identifiers, ignore rest
+        List<SampleIdentifier> unfilteredSampleIdentifiers = filterByStudyAndSample(studyViewFilter);
+
+        List<String> unfilteredStudyIds = new ArrayList<>();
+        List<String> unfilteredSampleIds = new ArrayList<>();
+        List<String> unfilteredPatientIds = new ArrayList<>();
+        List<String> studyIdsOfUnfilteredPatients = new ArrayList<>();
+        List<String> unfilteredUniqueSampleKeys = new ArrayList<>();
+        List<String> unfilteredUniquePatientKeys = new ArrayList<>();
+        List<String> unfilteredSampleAttributeIds = new ArrayList<>();
+        List<String> unfilteredPatientAttributeIds = new ArrayList<>();
+        // patient attributes which are also sample attributes in other studies
+        List<String> unfilteredConflictingPatientAttributeIds = new ArrayList<>();
+
+        populateIdLists(
+            // input
+            unfilteredSampleIdentifiers,
+            attributeIds,
+
+            // output
+            unfilteredStudyIds,
+            unfilteredSampleIds,
+            unfilteredPatientIds,
+            studyIdsOfUnfilteredPatients,
+            unfilteredUniqueSampleKeys,
+            unfilteredUniquePatientKeys,
+            unfilteredSampleAttributeIds,
+            unfilteredPatientAttributeIds,
+            unfilteredConflictingPatientAttributeIds
+        );
+
+        Map<String, ClinicalDataType> attributeDatatypeMap = constructAttributeDataMap(
+            unfilteredSampleAttributeIds,
+            unfilteredPatientAttributeIds,
+            unfilteredConflictingPatientAttributeIds
+        );
+
+        List<ClinicalData> unfilteredClinicalDataForSamples = clinicalDataFetcher.fetchClinicalDataForSamples(
+            unfilteredStudyIds,
+            unfilteredSampleIds,
```

Comment:
Since we are fetching data for all sample, can't we just use study ids

Reply:
Not sure if it is okay to ignore sample ids because, even if we call it unfiltered, actually we still filter by `sampleIdentifiers` here: https://github.com/onursumer/cbioportal/blob/f5e85a8c33181a8d13f166e36cab05118299945b/web/src/main/java/org/cbioportal/web/util/ClinicalDataBinUtil.java#L49

I thought we were doing this for a reason so didn't want to change the existing logic.